Fastapi Refactor

Fireye04 opened the pull request:
TODO:
- [x] Update current call method to make use of request body to provide parameters
- [x] Split endpoint configuration by source (similar to current click implementation)
- [x] Add error handling
- [ ] Add authorization?

Fireye04 committed:
init Fastapi

Fireye04 committed:
Functional FastAPI entrypoint for earthquake data

Fireye04 committed:
Run make update

Fix import bug

Fireye04 committed:
Post request with body

afausti commented on src/sasquatchbackpack/fastapi.py:
```diff
@@ -1,44 +1,47 @@
+    method: str
+
+
+@app.post("/sources/usgs/earthquake_data/")
```
A small suggestion here, it's best practice to use hyphens (-) instead of underscores (_) to separate words in URLs. 
Since `-data` doesn't add much consider just `/sources/usgs/earthquake`

afausti commented on src/sasquatchbackpack/fastapi.py:
```diff
@@ -1,44 +1,47 @@
-    method: str,
-) -> None:
+class EarthquakeParams(BaseModel):
+    """{days}/{hours}/{radius}/{latitude}/{longitude}/{lower}/{upper}/{publish}/{method}."""
```
Since this model represents the request body for querying earthquake events from the USGS API, you could name it `EarthquakeRequest` and also add a nice docstring

```
A request to query Earthquake events from the USGS API.

This model represents the request body for que earthquake query parameters.
```

afausti reviewed (changes requested):
This looks great, I've made a couple of suggestions mostly to improve docstrings and parameter descriptions in your model. 
Next we can talk about deploying this initial service and expanding the API with other endpoints.

Fireye04 commented:
Awesome! Do we want to directly integrate the functionality in fastAPI or are we ok with keeping the click dependency (and CLI entrypoint)

Fireye04 committed:
Polish Fastapi implementation
    add substantive docstrings
    add field defaults
    add field descriptions
    rename endpoint

Fireye04 commented:
Default args added! Now testable via
```
curl -X 'POST' \
  'http://localhost:8000/sources/usgs/earthquake/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "days": 20,
  "hours": 0
}'
```

afausti reviewed (approved):
Thanks for the fixes,  and yes we can keep the CLI as well.

Fireye04 committed:
Reorganize and add error handling

Fireye04 committed:
Init endpoint docs

Fireye04 committed:
Fix error handling

Fireye04 committed:
Progress api call docs

Fireye04 committed:
Run make update

Fireye04 committed:
Add testing to docs

Fireye04 committed:
Run make update

Fireye04 committed:
Update dockerfile to run fastAPI server

Fireye04 committed:
Docs fixes

Fireye04 committed:
Run Make Update

Fireye04 committed:
make updates

Fireye04 committed:
UV Refactor

Fireye04 committed:
Merge branch 'fix-tests' into tickets/fastapi

Fireye04 commented:
Ok had some UV problems, most of it should be fixed now, and the build system is mostly working again. Still need to figure out tox-docker though, might just temporarily disable that test so I can just get back to actual work.

Fireye04 committed:
Pause tests temporarily

Fireye04 committed:
Dockerfile udpates

Fireye04 committed:
unify python versions

Fireye04 committed:
Add gdal env

Fireye04 committed:
fixup

Fireye04 committed:
Newer debian image

Fireye04 committed:
Pin GDAl

Fireye04 committed:
Change distro

Fireye04 committed:
f

Fireye04 committed:
Update gdal pin

Fireye04 committed:
Try containers

Fireye04 committed:
test

Fireye04 committed:
test

Fireye04 committed:
Run make Update

Fireye04 committed:
reset

Fireye04 committed:
CI update

Fireye04 committed:
test

Fireye04 committed:
test

Fireye04 committed:
maybe?

Fireye04 committed:
run as user

Fireye04 committed:
e

Fireye04 committed:
test

Fireye04 committed:
uasfasd
